packages/vats/src/core/basic-behaviors.js

```diff
@@ -225,9 +224,39 @@ export const makeAddressNameHubs = async ({
 harden(makeAddressNameHubs);
 
 /** @param {BootstrapSpace} powers */
-export const makeClientBanks = async ({ consume: { client, bankManager } }) => {
+export const makeClientBanks = async ({
+  consume: {
+    agoricNames,
+    board,
+    namesByAddress,
+    namesByAddressAdmin,
+    client,
+    chainStorage,
+    bankManager,
+    zoe,
+  },
+  installation: {
+    consume: { singleWallet },
+  },
+}) => {
+  const STORAGE_PATH = 'wallet';
+
+  const storageNode = await getChildNode(chainStorage, STORAGE_PATH);
+  const marshaller = E(board).getPublishingMarshaller();
```

Comment:
post-script: wallets have state that shouldn't be published, so using the publishing marshaller was probably not a good idea. even using the read-only marshaller would be iffy: purses shouldn't be exposed to the board vat at all.

cc @warner

Reply:
The marshaller privileges and what is published are independent, no?

The latter is defined here https://github.com/Agoric/agoric-sdk/pull/5701/files#diff-b1c2b2b7ded05977f142adcd00946c70016bae3a5178a46af30ef444422e3332R60-R70

Which of those shouldn't be published? `purses`?

Are are you saying there should be separate streams for what goes to off-chain storage and what goes to the creator?

Reply:
> The marshaller privileges and what is published are independent, no?

Yes.

And indeed, purses must not be published. Nor payments. Probably not offers either. Publishing issuers and dapps is probably harmless. contacts is iffy...

But not only should we not publish purses, we shouldn't use the board's readOnlyMarshaller, since that involves exposing the purses to the board and trusting the board not to publish them.